### PR TITLE
Rewrite np.percentile section to handle units for AstroPy 3, 4.

### PR DESCRIPTION
**Pull request by wmwv**

### CODE REVIEW

File: python/lsst/validate/drp/calcsrd/adx.py [hunk 1]
```diff
@@ -91,9 +91,23 @@ def measureADx(metric, amx, afx_spec):
...
-            100. - afx_spec.threshold.value) * u.marcsec
+        #
+        # The AFx spec is in terms of percentage of outliers
+        # To convert to the percentitle of the largest outliers
+        # We subtract AFx from 100.  So AFx = 10 becomes threshold = 90.
+        threshold = 100 - afx_spec.threshold.value
+
+        # In AstroPy 3, numpy.percentitle of an array with quantities
+        #   (incorrectly) returns a unitless number
+        # In AstroPy 4, numpy.percentitle of an array with Quantities
```
Comment (timj) [inline python/lsst/validate/drp/calcsrd/adx.py]: I think it's Astropy4+numpy >=1.17.

File: python/lsst/validate/drp/calcsrd/adx.py [hunk 2]
```diff
@@ -91,9 +91,23 @@ def measureADx(metric, amx, afx_spec):
...
+        #   correct returns a number with the same units as the input array.
```
Comment (timj) [inline python/lsst/validate/drp/calcsrd/adx.py]: correct -> correctly

**Review by timj — APPROVED** [review]
Looks great. Nice clarification as well.

File: python/lsst/validate/drp/calcsrd/adx.py [hunk 3]
```diff
@@ -91,9 +91,23 @@ def measureADx(metric, amx, afx_spec):
...
+        # To be compatible with both Astropy 3.2 and Astropy 4, we:
+        # 1. Extract the array in marcsec and convert to a value.
+        # 2. Calculate the numpy.percentitle of this unitless array
+        # 3. Multiply that result by u.marcsec to get the right units.
+        rmsDistMas = amx.extras['rmsDistMas'].quantity.to(u.marcsec).value,
```
Comment (timj) [inline python/lsst/validate/drp/calcsrd/adx.py]: There's a comma at the end of this line that should not be there.

**Comment by timj** [comment]
I also needed to apply this patch:
```diff
diff --git a/python/lsst/validate/drp/calcsrd/pa2.py b/python/lsst/validate/drp/calcsrd/pa2.py
index 6dd6c0a..8b98c47 100644
--- a/python/lsst/validate/drp/calcsrd/pa2.py
+++ b/python/lsst/validate/drp/calcsrd/pa2.py
@@ -59,5 +59,5 @@ def measurePA2(metric, pa1, pf1_thresh):
     magDiffs = pa1.extras['magDiff'].quantity[0, :]
 
     pf1Percentile = 100.*u.percent - pf1_thresh
-    return Measurement(metric, np.percentile(np.abs(magDiffs), pf1Percentile.value) * magDiffs.unit,
+    return Measurement(metric, np.percentile(np.abs(magDiffs.value), pf1Percentile.value) * magDiffs.unit,
                        extras=datums)
```

**Comment by wmwv** [comment]
The one other use of `numpy.percentile` I found looks like

https://github.com/lsst/validate_drp/blob/7397373f0857cefcbb27b649c4ea122956b5991a/python/lsst/validate/drp/repeatability.py#L315

I think that one is fine as is.

**Pull request merged**